src/components/tooltips-hoc/TooltipHOC.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import {DropPod} from './DropPod';
+
+export interface ITooltipHOCProps {
+    buttonContainer?: React.HTMLAttributes<HTMLDivElement>;
+    listContainer?: React.HTMLAttributes<HTMLDivElement>;
+    renderOpenButton: (onClick: () => void) => React.ReactNode;
+    closeOnClick?: boolean;
+    positions?: string[];
+    selector?: string;
+}
+
+export interface ITooltipHOCState {
+    isOpen: boolean;
+}
+
+export class TooltipHOC extends React.PureComponent<ITooltipHOCProps, ITooltipHOCState> {
+    readonly button: React.RefObject<HTMLDivElement>;
+
+    static defaultProps: Partial<ITooltipHOCProps> = {
+        closeOnClick: true,
+    };
+
+    constructor(props: ITooltipHOCProps, state: ITooltipHOCState) {
+        super(props, state);
+
+        this.button = React.createRef();
+        this.state = {
+            isOpen: false,
+        };
+    }
+
+    componentWillMount() {
+        document.addEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    render() {
+        return (
+            <>
+                <div

```

Comment:
Do you really need a div or you could simply put 
```typescript
render() {
  return (
    <>
      {this.props.renderOpenButton(() => this.onClick())}
      {this.createPortalMenu()}
    </>
  );
```

Reply:
I need my ref. Its necessary to calcul the offset